src/ray/raylet/scheduling/cluster_resource_data.h

```diff
@@ -237,3 +398,17 @@ NodeResources ResourceMapToNodeResources(
 TaskRequest ResourceMapToTaskRequest(
     StringIdMap &string_to_int_map,
     const std::unordered_map<std::string, double> &resource_map);
+
+namespace std {
+template <>
+struct hash<ResourceSet> {
+  size_t operator()(ResourceSet const &k) const {
+    size_t seed = k.GetResourceMap().size();
```

Comment:
1) better cache the result.
2) this variable should be named `hash`?

Reply:
This is copied from the old data structure, let's keep it as original in this PR to avoid potential bugs. We can open another PR to do it after this PR is merged.